scripts/gb-model/retrieve_unavailability_data.py

```diff
@@ -0,0 +1,756 @@
+
+                            # Calculate unavailable capacity
+                            if outage_data["nominal_power_mw"]:
+                                period_data["min_unavailable_mw"] = max(
+                                    0, outage_data["nominal_power_mw"] - max(quantities)
+                                )
+                                period_data["max_unavailable_mw"] = max(
+                                    0, outage_data["nominal_power_mw"] - min(quantities)
+                                )
+
+                    outages.append(period_data)
+            else:
+                # No Available_Period elements, just add the TimeSeries data
+                outages.append(outage_data)
+
+    except ET.ParseError as e:
+        logger.error(f"Failed to parse XML {filename}: {e}")
+    except Exception as e:
+        logger.error(f"Error processing XML {filename}: {e}")
+
+    return outages
+
+
+def _get_text(element, xpath: str, namespaces: dict[str, str]) -> Optional[str]:
+    """
+    Helper function to safely extract text from XML element
+
+    Args:
+        element: XML element
+        xpath: XPath expression
+        namespaces: XML namespaces
+
+    Returns:
+        Text content or None
+    """
+    found = element.find(xpath, namespaces)
+    return found.text if found is not None else None
+
+
+def process_unavailability_data(
+    df: pd.DataFrame, output_file: str, bidding_zone: str, config: dict
+) -> None:
+    """
+    Process and save unavailability data
+
+    Args:
+        df: DataFrame with raw unavailability data
+        output_file: Path to save processed data
+        bidding_zone: Bidding zone identifier
+        config: Snakemake configuration dictionary
+    """
+    if df.empty:
+        logger.warning("No data to process")
+        # Create empty CSV with headers
+        empty_df = pd.DataFrame(
+            columns=[
+                "resource_mrid",
+                "resource_name",
+                "resource_location",
+                "resource_type",
+                "business_type",
+                "start_time",
+                "end_time",
+                "nominal_power_mw",
+                "min_available_mw",
+                "max_available_mw",
+                "avg_available_mw",
+                "min_unavailable_mw",
+                "max_unavailable_mw",
+                "bidding_zone",
+                "business_type_desc",
+            ]
+        )
+        empty_df.to_csv(output_file, index=False)
+        return
+
+    # Add bidding zone information
+    df["bidding_zone"] = bidding_zone
+
+    # Clean and process data
+    entso_e_config = config.get("unavailability", {}).get("entso_e", {})
+    business_type_map = {
+        entso_e_config.get("business_types", {}).get(
+            "planned", "A53"
+        ): "Planned maintenance",
+        entso_e_config.get("business_types", {}).get(
+            "forced", "A54"
+        ): "Forced unavailability",
```

Comment:
you define forced == A54 and planned == A53 in the snakemake config so you should use that mapping again here.